helm: Use variable for spire server config container image

eegseth opened the pull request:
Previously, the busybox image used for configuring spire server was hardcoded into the statefulset.

This patch moves this image reference to the variable authentication.mutual.spire.install.server.spireConfig.image

Fixes: cilium#27282

Signed-off-by: E Egseth <endre@egseth.no>

eegseth committed:
Previously, the busybox image used for configuring spire server was hardcoded into the statefulset.

This patch moves this image reference to the variable authentication.mutual.spire.install.server.spireConfig.image

Fixes: cilium#27282

Signed-off-by: E Egseth <endre@egseth.no>

eegseth committed:
Previously, the busybox image used for configuring spire server and agent was hardcoded into the statefulset.

This patch moves this image reference to the variable authentication.mutual.spire.install.spireConfig.image

Fixes: #27282

Signed-off-by: E Egseth endre@egseth.no

squeed commented on install/kubernetes/cilium/values.yaml.tmpl:
```diff
@@ -3141,6 +3141,9 @@ authentication:
         # -- SPIRE namespace to install into
         namespace: cilium-spire
+        # -- SPIRE configuration image
+        spireConfig:
```
nit -- this should be called `init`, since it matches the container name.

@meyskens can we just use spire-agent? does it have a shell?

eegseth replied:
Renamed. Looks like the spire agent doesen't have a shell.

eegseth committed:
Previously, the busybox image used for configuring spire server and agent was hardcoded into the statefulset.

This patch moves this image reference to the variable authentication.mutual.spire.install.init.image

Fixes: cilium#27282

Signed-off-by: E Egseth endre@egseth.no

eegseth committed:
Merge branch 'main' into eegseth/spire-server-config-container-img-as-var

Signed-off-by: Endre Egseth <endre@egseth.no>

eegseth committed:
docs: update with new helm var for spire init container.

Signed-off-by: E Egseth endre@egseth.no
